grapple/helpers.py

```diff
@@ -217,3 +213,61 @@ def resolve_plural(self, _, info, **kwargs):
         return cls
 
     return inner
+
+
+def register_singular_query_field(field_name, query_params=None, required=False):
+    def inner(cls):
+        field_type = lambda: registry.models[cls]
+        field_query_params = query_params or {
+            "token": graphene.Argument(
+                graphene.String, description=ugettext_lazy("The preview token.")
+            ),
+            "order": graphene.Argument(
+                graphene.String,
+                description=ugettext_lazy("Use the Django QuerySet order_by format."),
+            ),
+        }
+
+        def Mixin():
+            # Generic methods to get all and query one model instance.
+            def resolve_singular(self, _, info, **kwargs):
+                try:
+                    # If is a Page then only query live/public pages.
+                    if issubclass(cls, Page):
+                        if "token" in kwargs and hasattr(
+                            cls, "get_page_from_preview_token"
+                        ):
+                            return cls.get_page_from_preview_token(kwargs["token"])
+
+                        qs = cls.objects.live().public()
+                        if "order" in kwargs:
+                            qs = qs.order_by(kwargs.pop("order"))
+                        return qs.filter(**kwargs).first()
+
+                    return cls.objects.first()
```

Comment:
The `order` argument isn't being used here.
Updating to `return cls.objects.filter(**kwargs).first()` should do the trick.

Reply:
nice catch. sorted!

also changed to only add the default token param for `Page` subclasses in d28fa5b